src/App.js

```diff
@@ -121,40 +103,48 @@ class App extends Component {
 
   render() {
     return (
-      <main className="App">
-      {this.state.user ?
-        <Header 
-          buttonDisplay={this.logOut} 
-          user={this.state.user}
-          buttonText='Log Out'
-          title={`Welcome ${this.state.user.name}`}
-          /> :
-        <Header 
-          buttonDisplay={this.showLoginPage}
-          buttonText='Log In'
-          title='Welcome to Rancid Tomatillos'
-          />
-      }
-        {this.state.login &&
-          <Login
-            fetchUserData={this.fetchUserData} 
-            closeLoginPage={this.closeLoginPage}/>}
-        {this.state.homepage &&
-          <section className="home-page">
-            <CardContainer 
-              movies={this.state.movies} renderSpecificMovie={this.renderSpecificMovie}/>
-          </section>
-        }
-        {this.state.selectedMovie &&
-            <Movie 
-              movie={this.state.selectedMovie}
-              closeMovieDetail={this.closeMovieDetail}
+      <BrowserRouter>
+        <main className="App">
+        {this.state.user ?
+          <Header 
+            buttonDisplay={this.logOut} 
+            user={this.state.user}
+            buttonText='Log Out'
+            title={`Welcome ${this.state.user.name}`}
+            /> :
+          <Header 
+            buttonDisplay={this.showLoginPage}
+            buttonText='Log In'
+            title='Welcome to Rancid Tomatillos'
```

Comment:
Opportunity for refactor here. The logic of displaying info based on whether or not the user is present should be in the Header itself. Instead of rendering two different headers, render one header with three props: user, showLoginPage, and logOut. In the Header itself, the logic for deciding which text to display and which function to run on button click, should be determined by whether or not the user prop exists or not.

Reply:
I agree 100%.